File: api/clip.f90
program Clipit
  use rsf

  implicit none
  type (file)                      :: in, out
  integer                          :: n1, n2, i1, i2
  real                             :: clip
  real, dimension (:), allocatable :: trace

  call sf_init()
  in = rsf_input()
  out = rsf_output()

  call from_par(in,"n1",n1)
  n2 = filesize(in,1)

  call from_par("clip",clip)

  allocate (trace (n1))

  do i2=1, n2
     call rsf_read(in,trace)
     
     where (trace >  clip) trace =  clip
     where (trace < -clip) trace = -clip

     call rsf_write(out,trace)
  end do
end program Clipit

File: api/clip.cc
#include <valarray>

#include <rsf.hh>

int main(int argc, char* argv[])
{
    sf_init(argc,argv);

    iRSF par(0), in;
    oRSF out;

    int n1, n2;

    in.get("n1",n1);
    n2=in.size(1);

    float clip;

    par.get("clip",clip);

    std::valarray<float> trace(n1);

    for (int i2=0; i2 < n2; i2++) {
	in >> trace;

	for (int i1=0; i1 < n1; i1++) {
	    if      (trace[i1] >  clip) trace[i1]=clip;
	    else if (trace[i1] < -clip) trace[i1]=-clip;
	}

	out << trace;
    }

    exit(0);
}
